packages/ui-extensions/src/surfaces/admin/api/resource-picker/resource-picker.ts

```diff
@@ -0,0 +1,240 @@
+  storefrontId: string;
+  templateSuffix?: string | null;
+  title: string;
+  updatedAt: string;
+}
+
+interface ProductVariant extends Resource {
+  availableForSale: boolean;
+  barcode?: string | null;
+  compareAtPrice?: Money | null;
+  createdAt: string;
+  displayName: string;
+  fulfillmentService?: {
+    id: string;
+    inventoryManagement: boolean;
+    productBased: boolean;
+    serviceName: string;
+    type: FulfillmentServiceType;
+  };
+  image?: Image | null;
+  inventoryItem: {id: string};
+  inventoryManagement: ProductVariantInventoryManagement;
+  inventoryPolicy: ProductVariantInventoryPolicy;
+  inventoryQuantity?: number | null;
+  position: number;
+  price: Money;
+  product: Partial<Product>;
+  requiresShipping: boolean;
+  selectedOptions: {value?: string | null}[];
+  sku?: string | null;
+  taxable: boolean;
+  title: string;
+  weight?: number | null;
+  weightUnit: WeightUnit;
+  updatedAt: string;
+}
+
+interface Product extends Resource {
+  availablePublicationCount: number;
+  createdAt: string;
+  descriptionHtml: string;
+  handle: string;
+  hasOnlyDefaultVariant: boolean;
+  images: Image[];
+  options: {
+    id: string;
+    name: string;
+    position: number;
+    values: string[];
+  }[];
+  productType: string;
+  publishedAt?: string | null;
+  tags: string[];
+  templateSuffix?: string | null;
+  title: string;
+  totalInventory: number;
+  totalVariants: number;
+  tracksInventory: boolean;
+  variants: Partial<ProductVariant>[];
+  vendor: string;
+  updatedAt: string;
+  status: ProductStatus;
+}
+
+interface ResourceTypes {
+  product: Product;
+  variant: ProductVariant;
+  collection: Collection;
+}
+
+type ResourceSelection<Type extends keyof ResourceTypes> = ResourceTypes[Type];
+
+type WithSelection<T> = T & {
+  /**
+   * @private
+   * @deprecated
+   */
+  selection: T;
+};
+
+type SelectPayload<Type extends keyof ResourceTypes> = WithSelection<
+  ResourceSelection<Type>[]
+>;
+
+interface Filters {
+  /**
+   * Whether to show hidden resources, referring to products that are not published on any sales channels.
+   * @defaultValue true
+   */
+  hidden?: boolean;
+  /**
+   * Whether to show product variants. Only applies to the Product resource type picker.
+   * @defaultValue true
+   */
+  variants?: boolean;
+  /**
+   * Whether to show [draft products](https://help.shopify.com/en/manual/products/details?shpxid=70af7d87-E0F2-4973-8B09-B972AAF0ADFD#product-availability).
+   * Only applies to the Product resource type picker.
+   * Setting this to undefined will show a badge on draft products.
+   * @defaultValue true
+   */
+  draft?: boolean | undefined;
+  /**
+   * Whether to show [archived products](https://help.shopify.com/en/manual/products/details?shpxid=70af7d87-E0F2-4973-8B09-B972AAF0ADFD#product-availability).
+   * Only applies to the Product resource type picker.
+   * Setting this to undefined will show a badge on draft products.
```

Comment:
I think it is supposed to be draft since you can set the product status to draft.
<img width="992" alt="Screenshot 2024-06-10 at 3 40 40 PM" src="https://github.com/Shopify/ui-extensions/assets/390744/b0fce830-2dda-4010-9eb2-2dcfa86a059f">

Reply:
Ahh, gotcha 👍